pkg/runtime/reconciler.go

```diff
@@ -604,27 +643,29 @@ func (r *resourceReconciler) patchResourceMetadataAndSpec(
 	}
 
 	rlog.Enter("kc.Patch (metadata + spec)")
-	// Save a copy of the latest object, to reset 'Status' after performing
-	// the kc.Patch() operation
-	latestCopy := latest.DeepCopy()
-	err = r.kc.Patch(
-		ctx,
-		latest.RuntimeObject(),
-		client.MergeFrom(desired.DeepCopy().RuntimeObject()),
-	)
-	// Reset the status of latest object after patching.
-	latest.SetStatus(latestCopy)
-	rlog.Exit("kc.Patch (metadata + spec)", err)
-
-	if err != nil {
-		return err
+	dobj := desired.DeepCopy().RuntimeObject()
+	lorig := latest.DeepCopy()
+	patch := client.MergeFrom(dobj)
+	err = r.kc.Patch(ctx, latest.RuntimeObject(), patch)
+	if err == nil {
+		if rlog.IsDebugEnabled() {
+			js := getPatchDocument(patch, lorig.RuntimeObject())
+			rlog.Debug("patched resource metadata + spec", "json", js)
+		}
 	}
-	rlog.Debug("patched resource metadata and spec", "latest", latest)
-	return nil
+	// The call to Patch() above ends up setting the latest variable's Status
+	// to the value of the desired variable's Status. We do not want this
+	// behaviour; instead, we want to keep latest's original Status value.
+	latest.SetStatus(lorig)
```

Comment:
@RedbackThomson I reverted this back to the original code (see this line above).